### DISCUSSION TITLE
Window Detection - window state icon missing

### DISCUSSION THREAD

**Discussion by pflugj** [jmcollin78/versatile_thermostat#570] [General]
Hi.

The "window detection" is enabled and working ;-)

I am sure, that at one stage,
there was an **icon indicating the window state** ( Open / close)
in the "Versatile Thermostat Climate Card"

as far as I remember it showed somewhere here:
![image](https://github.com/user-attachments/assets/9c3a884c-faaf-45b6-9dfc-614ddb78e777)

but it seems to have gone .
The windows state is showing in the "sensors":
![image](https://github.com/user-attachments/assets/446f30d4-f60b-42e0-8095-e7bf0dc74ae0)

I'm tried:
1. window detection disabled then enabled
2. remove / add the window sensor 
3. Removed added the Thermostat and the Climate Card

I can actually add  a tile and display the "window state sensor "
![image](https://github.com/user-attachments/assets/de8813d5-ed8e-4fae-ae4c-d0e5c1ca8850)

And I expect:
An Icon of the door state being displayed in the Thermostat card

But I observe this ....
NO "window state"  Icon :-(

I am not sure if this is a bug ore is that icon gone?? 
Any idea???

Thanks for helping
.... and a brilliant Thermostat implementation

**Comment by jmcollin78** [comment]
Hello @pflugj ,

I just try at home and it is working fine:
![Capture d’écran 2024-10-26 à 10 07 23](https://github.com/user-attachments/assets/29e4d9d8-850c-4526-8469-322e1562910b)

I have no idea of what happens in your case. The VTherm UI Card use the attribute window_state to set the icon and in your case, this attribute is really 'on'. 
```
if (attributes?.window_state === 'on' || attributes?.window_auto_state === 'on') {
          this._hasWindow = true;
          this.window = true;
        }
```

Can you check in the Development Tools / States what is the value of the `window_state` attribute. It should be 'on' or 'off'.

**Reply by pflugj** [reply]
Hi. 
I did as requested and got this:
![image](https://github.com/user-attachments/assets/36f6f4cb-90de-4c54-a0a0-2b670a0de981)
I can change the state manually

![image](https://github.com/user-attachments/assets/6983b02f-5dce-4094-add7-c2fb046d5dc8)
(All doors and windows closed)
I am using:
A) a binary senor:  "Office Windows" 
B) a door sensor:  "Balcony Door"
those are combined into 3rd binary sensor
C) "Office Door & Windows"
This is the one I use for the window detection in VTherm
the 4th one is the one provided by VTherm:
D) binary_sensor.ac_office_window_state = "Window State"
![image](https://github.com/user-attachments/assets/8cd52fae-1d16-4b1e-82d2-8c82a13fc5f6)
 I did change the "Window State" manually in Development Tools / States
 That is why the the time is only "21 minutes ago"

I the changed the "window_state" manually to "ON" and than I got this:
![image](https://github.com/user-attachments/assets/0460fd97-80b0-41cf-abbd-0884cc0f1f34)

I just noted that the "comfort" icon is also not displayed.
The settings are there:
![image](https://github.com/user-attachments/assets/95357bb9-e042-4089-b1be-faa3714322ac)
Are there any other Icons that can be displayed??

... and YES it is that hot here :-)

Thanks for helping
Cheers
Jurgen

**Reply by jmcollin78** [reply]
The correct state is 'on' and not 'ON'. Give it a try please

**Comment by pflugj** [comment]
I do have a 2nd AC (without window detection)
Which looks like this:
![image](https://github.com/user-attachments/assets/3296e16f-1116-4120-82fc-96d6b67144f5)

What do this icon mean?

**Reply by jmcollin78** [reply]
1. the power of the device configured
2. the self-regulation algorithm configured,
3. the fan should be 'on'.

Note that you can turn off all this complementary infos.

**Comment by pflugj** [comment]
Well, as I have done before, 
I added the card again 'et voilà', the icons are back :-)
side by side 
![image](https://github.com/user-attachments/assets/9cf4fe69-cf38-4f7b-aaf9-2e23465ed525)
- weird - even the temperature is different

Just some clarification:
![image](https://github.com/user-attachments/assets/9d86b179-b441-452f-9b59-d85adc5839a1)
The power is just a guess ( I believe)??
(The real power is measured by the switch that powers the AC and displayed in the picture above)

![image](https://github.com/user-attachments/assets/6f925694-4cee-4076-afcf-e9b5ec8cd28e)
Means that the FAN is set to "Auto HIGH" 
 
![image](https://github.com/user-attachments/assets/5cd88bb6-052f-4ab7-b697-06d2311e5b77)
Means what ??

I have been reading through the HUGE manual, but I missed where I can disable the complementary settings :-(

Thanks again for helping to understand this incredible complex Thermostat
Cheers
Jurgen

**Reply by jmcollin78** [reply]
> Just some clarification:
> ![image](https://private-user-images.githubusercontent.com/17050757/380468470-9d86b179-b441-452f-9b59-d85adc5839a1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMTc3MTUsIm5iZiI6MTczMDAxNzQxNSwicGF0aCI6Ii8xNzA1MDc1Ny8zODA0Njg0NzAtOWQ4NmIxNzktYjQ0MS00NTJmLTliNTktZDg1YWRjNTgzOWExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDA4MjMzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkMTBhNTgzMDU4ZmRiNjU1ZmYwN2RjOGZlOWU1ZWVlMzJkN2M3MDdhOWQyOTgyNjEyMjVjMDZlOGRmYTNmNjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1xusERDzVW_ZX0Q1TRh-YQY71c6Hp_vLRNgsor6OjzI)
> The power is just a guess ( I believe)??

The power is based on your have configured for the VTherm. If real consumption is 1.9 kw you should rather configure 2 kw for this VTherm. This is configured in the "Main attributes" page:

> ![Capture d’écran 2024-10-27 à 09 28 08](https://github.com/user-attachments/assets/f0028966-7452-4dd1-8492-03d4c00a71ed)

Yes


> ![image](https://github.com/user-attachments/assets/5cd88bb6-052f-4ab7-b697-06d2311e5b77)
> Means what ??

This is the current fan mode. In your case, fan is 'on'.

**Comment by pflugj** [comment]
Again - missed that power-setting :-(
In the picture, it was actually 1.9W :-) 

The newer AC's are using inverters and do not have a fixed power.
Mine start at ~100W and go up to 1.3KW

Meaning In my case the setting is not really useful.
I switched the Icon off by using a "0" entry 
(a blank entry was not accepted and reverted to "1")

The later AC's do actually deliver the power-consumption themself.
I am still using switches with metering function.

Maybe a new feature??
Retrieve the actual consumption from another entity??

**Reply by jmcollin78** [reply]
If you have the power from another entity why do you to integrate them in Vtherm ?
In Vtherm this is for people that don't have such an entity and want something near the real value.
